metadata_backend/api/handlers.py

```diff
@@ -262,13 +262,54 @@ async def put_object(self, req: Request) -> Response:
         return web.Response(body=body, status=201,
                             content_type="application/json")
 
+    async def patch_object(self, req: Request) -> Response:
+        """Update metadata object in database.
+
+        We do not support patch for XML.
+
+        :param req: PATCH request
+        :returns: JSON response containing accessionId for submitted object
+        """
+        schema_type = req.match_info['schema']
+        accession_id = req.match_info['accessionId']
+        self._check_schema_exists(schema_type)
+        collection = (f"draft-{schema_type}" if req.path.startswith("/drafts")
+                      else schema_type)
+
+        db_client = req.app['db_client']
+        operator: Union[Operator, XMLOperator]
+        if req.content_type == "multipart/form-data":
+            reason = "XML patching is not possible."
+            raise web.HTTPUnsupportedMediaType(reason=reason)
+        else:
+            content = await req.json()
+            operator = Operator(db_client)
+        await operator.update_metadata_object(collection,
+                                              accession_id,
+                                              content)
+        body = json.dumps({"accessionId": accession_id})
+        LOG.info(f"PUT object with accesssion ID {accession_id} "
+                 f"in schema {collection} was successful.")
+        return web.Response(body=body, status=201,
+                            content_type="application/json")
+
     async def get_folders(self, req: Request) -> Response:
         """Get all possible object folders from database.
 
         :param req: GET Request
-        :returns: JSON list of folders
+        :returns: JSON list of folders available for the user
         """
-        raise web.HTTPNotImplemented
+        db_client = req.app['db_client']
+        db_service = DBService('folders', db_client)
+        collection = db_service.database['folder']
```

Comment:
We can use query method from db_services here and then collect folders with list comprehension:

```
cursor = db_service.query("folder", {})
folders = [folder async for folder in cursor]
```

Reply:
this was a nice fix 👍